File: ws/cañarte/u2/WS24Zoo/ZOO/src/main/java/ec/edu/espe/zoo/model/Fish.java
package ec.edu.espe.zoo.model;

import java.util.Date;

/**
 *
 * @author David Morillo, JavaSquad, DCCO-ESPE
 */
public abstract class Fish extends Animal{
    
    public Fish(int id, String description, Date dateOnborn, char gender, int numberOfBones, Cage cage) {
        super(id, description, dateOnborn, gender, numberOfBones, cage);
    }
    
}

File: hw/Vaca/u2/HW18ZooInPythonJavaScript/Zoo/ec.edu.espe.zoo.model/Fish.js
const Animal = require('./Animal');

class Fish extends Animal {
    constructor(id, description, dateOnborn, gender, numberOfBones, cage, waterType) {
        super(id, description, dateOnborn, gender, numberOfBones, cage);
        this.waterType = waterType;
    }

    getWaterType() {
        return this.waterType;
    }

    setWaterType(waterType) {
        this.waterType = waterType;
    }

    toString() {
        return `Fish{waterType=${this.waterType}${super.toString()}}`;
    }
}

module.exports = Fish;
